src/python/foglamp/update_statistics_history.py

```diff
@@ -0,0 +1,139 @@
+                                sqlalchemy.Column('value', sqlalchemy.BIGINT, default=0),
+                                sqlalchemy.Column('previous_value', sqlalchemy.BIGINT, default=0),
+                                sqlalchemy.Column('ts', sqlalchemy.TIMESTAMP(6), default=
+                                time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(time.time()))))
+
+"""Description of each column 
+key - Corresponding statistics.key value, so that there is awareness of what the history is of
+history_ts - the newest timestamp in statistics for that key 
+value - delta value between `value` and `prev_val` of statistics
+            The point is to know how many actions happened during the X time
+ts - current timestamp 
+"""
+_STATS_HISTORY_TABLE = sqlalchemy.Table('statistics_history', sqlalchemy.MetaData(),
+                                        sqlalchemy.Column('key', sqlalchemy.CHAR(10)),
+                                        sqlalchemy.Column('history_ts', sqlalchemy.TIMESTAMP(6), default=
+                                        time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(time.time()))),
+                                        sqlalchemy.Column('value', sqlalchemy.BIGINT, default=0),
+                                        sqlalchemy.Column('ts', sqlalchemy.TIMESTAMP(6), default=
+                                        time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(time.time())))
+                                        )
+
+
+def __list_stats_keys() -> list:
+    """
+    generate a list of distinct keys from statistics table 
+    Returns:
+        list of distinct keys
+    """
+    key_list = []
+    stmt = sqlalchemy.select([_STATS_TABLE.c.key.distinct()]).select_from(_STATS_TABLE)
+    result = _CONN.execute(stmt)
+    result = result.fetchall()
+    for i in range(len(result)):
+        key_list.append(result[i][0].replace(" ", ""))
+
+    return key_list
+
+# List of PRIMARY KEY values in statistics
+_STATS_KEY_VALUE_LIST = __list_stats_keys()
+
+
+def __insert_into_stats_history(key='', history_ts=time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(time.time())),
+                                value=0):
+    """
+    INSERT values in statistics_history
+    Args:
+        key: corresponding stats_key_value
+        history_ts: timestamp in statistics table for corresponding key 
+        value: delta between `value` and `prev_val`
+    Returns:
+        Return the number of rows inserted. Since each process inserts only 1 row, the expected count should always 
+        be 1. 
+    """
+    stmt = _STATS_HISTORY_TABLE.insert().values(key=key, history_ts=history_ts, value=value)
+    _CONN.execute(stmt)
+
+
+def __update_previous_value(key=''):
+    """Query: 
+    UPDATE 
+        statistics 
+    SET 
+        previous_value = (SELECT value FROM statistics WHERE key='READINGS')
+    WHERE 
+        value = (SELECT value FROM statistics WHERE key='READINGS'); 
+    """
```

Comment:
suppose key_arg and value_arg are parameters passed in to __update_previous_value, the logic should be:
UPDATE statistics SET previous_value = value_arg WHERE key = key_arg